src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy

```diff
@@ -442,6 +442,31 @@ class GitVersionPluginTests extends Specification {
         buildResult.output =~ ":printVersionDetails\n1.0.0\n"
     }
 
+    def 'git describe with commit after tag' () {
+        given:
+        buildFile << '''
+            plugins {
+                id 'com.palantir.git-version'
+            }
+            version gitVersion()
+        '''.stripIndent()
+        gitIgnoreFile << 'build'
+        Git git = Git.init().setDirectory(projectDir).call();
+        git.add().addFilepattern('.').call()
+        git.commit().setMessage('initial commit').call()
+        git.tag().setAnnotated(true).setMessage('1.0.0').setName('1.0.0').call()
```

Comment:
Is it worth verifying the code works for a non annotated tag too?

Reply:
Is that not tested already? If not, we should probably add it but it'd be independent of this change IMHO.

Reply:
Given that this PR touches the `%s-%s-g%s` code, I think it's worth proving that it works correctly when it encounters both annotated and non-annotated tags.  I couldn't see an existing test case that captures this - all the others just do `git.tag().setAnnotated(true)`.

If it turns out to require other code-changes to pass, then I'm happy to put that test in a separate PR.

Reply:
Ok sure, I'll add tests for both cases (the exact tag case and the %s-%s-g%s case).